optimade/models/structures.py

```diff
@@ -857,36 +885,33 @@ def check_periodic_dimensions(cls, v, values):
 
     @validator("lattice_vectors", always=True)
     def required_if_dimension_types_has_one(cls, v, values):
-        if (
-            Periodicity.PERIODIC.value in values.get("dimension_types", [])
-            and v is None
-        ):
-            raise ValueError(
-                f"lattice_vectors is REQUIRED, since dimension_types is not ({(Periodicity.APERIODIC.value,) * 3}) but is "
-                f"{tuple(getattr(_, 'value', None) for _ in values.get('dimension_types', []))}"
-            )
-
-        for dim_type, vector in zip(values.get("dimension_types", (None,) * 3), v):
-            if None in vector and dim_type == Periodicity.PERIODIC.value:
-                raise ValueError(
-                    f"Null entries in lattice vectors are only permitted when the corresponding dimension type is {Periodicity.APERIODIC.value}. "
-                    f"Here: dimension_types = {tuple(getattr(_, 'value', None) for _ in values.get('dimension_types', []))}, lattice_vectors = {v}"
-                )
+        if v and values.get("dimension_types"):
+            for dim_type, vector in zip(values.get("dimension_types", (None,) * 3), v):
+                if None in vector and dim_type == Periodicity.PERIODIC.value:
+                    raise ValueError(
+                        f"Null entries in lattice vectors are only permitted when the corresponding dimension type is {Periodicity.APERIODIC.value}. "
+                        f"Here: dimension_types = {tuple(getattr(_, 'value', None) for _ in values.get('dimension_types', []))}, lattice_vectors = {v}"
+                    )
 
         return v
 
     @validator("lattice_vectors")
     def null_values_for_whole_vector(cls, v):
-        for vector in v:
-            if None in vector and any((isinstance(_, float) for _ in vector)):
-                raise ValueError(
-                    f"A lattice vector MUST be either all `null` or all numbers (vector: {vector}, all vectors: {v})"
-                )
+        if v:
```

Comment:
You might argue that all these `if v: validate(...)` tests should be changed into `if v is None: return v (else:) validate()`...

Reply:
To follow the discussion from the just merged #547, then yes :)

Reply:
Hopefully resolved in my latest commit, as well as the re-introduction of validating assemblies discussed above